README.md

```diff
@@ -0,0 +1,179 @@
+bleanser or 'backup cleanser' is a tool for cleaning old and redundant backups
+
+In this context, backup typically means something like a GDPR export, an XML or JSON file which includes your data from some website/API, or a sqlite database from an application
+
+<https://beepb00p.xyz/exobrain/projects/bleanser.html>
+
+This is used to find 'redundant backups'. As an example, say you save your data to a JSON file by making API requests to some API service once a day. If your export of the data you exported today is a [superset](https://en.wikipedia.org/wiki/Subset) of the export yesterday, you know you can safely delete the old file and still have a complete backup of your data. This helps:
+
+- save on disk space
+- save of data access time; how long it takes to parse all your input files (see [data access layer](https://beepb00p.xyz/exports.html#dal))
+
+This works for both [full](https://beepb00p.xyz/exports.html#full) (you're able to get all your data from a service) and [incremental](https://beepb00p.xyz/exports.html#incremental) exports.
+
+This is especially relevant for incremental data exports, as they're harder to reason about. So, this handles the complex bits of diffing adjacent backups.
+
+As an example of an incremental export, imagine the service you were using only gave you access to the latest 3 items in your history (a real example of this is the [github activity feed](https://github.com/karlicoss/ghexport))
+
+| Day 1 | Day 2 | Day 3 |
+| ----- | ----- | ----- |
+| A     | B     | C     |
+| B     | C     | D     |
+| C     | D     | E     |
+
+To parse this in your [data access layer](https://beepb00p.xyz/exports.html#dal), you could imagine something like this:
+
+```python
+events = set()
+for file in inputs:
+    for line in file:
+        events.add(line)
+# events is now {'A', 'B', 'C', 'D', 'E'}
+```
+
+You might notice that if you removed 'Day 2', you'd still have an accurate backup, and we'd still have all 5 items, but its not obvious you can remove it since none of these are supersets of each other.
+
+`bleanser` is meant to solve this problem in a data agnostic way, so any export can be converted to a unique 'snapshot', and those can be sorted/compared against each other to find redundant data
```

Comment:
hmm, snapshot feels a bit confusing as well? Usually implies there is some process in time, whereas in this case the export we're processing is already a snapshot. Maybe "normalised representation" or something like that? Would also be consistent with the name for `Normaliser`? 
Could probably also rename `do_cleanup/cleanup` to `do_normalise/normalise`

Reply:
yeah, will make a separate issue for that I think, can keep backwards compatability by just having the `do_cleanup` in `BaseNormaliser` call `do_normalize` after sending a warning